### PR TITLE
Expose pre-training APIs in SFT and OSFT

### PR DESCRIPTION
**Pull request by RobotSail**
This PR exposes pre-training mode in the OSFT and SFT algorithms.

In order to use pretraining data, the user specifies `is_pretraining=True` to the `sft` and `osft` APIs. 

This tells the algorithms to treat the dataset as a pretraining dataset containing documents which are placed in the `document` column. 

The `block_size` argument controls the size of a singular pretraining sample (contiguous list of training data). 

The way data is processed works like this:

1. Each `document` in a dataset is tokenized as `tokenizer.encode(document) + tokenizer.eos`, then set as `input_ids`. 
2. The training backend loads this tokenized dataset and packs it into blocks of `block_size`. Each block is essentially just a slice from the original list of `block_size`. 

Otherwise, the training loop functions exactly the same as standard training. 

## Summary by CodeRabbit

* **New Features**
  * Added pretraining mode to SFT and OSFT: enable pretraining, set block size, and specify document field for raw-document training.

* **Bug Fixes / Validation**
  * Enforced block_size when pretraining is enabled; added mutual-exclusion checks and warnings for incompatible options.

* **Documentation**
  * New guides, examples, API docs, and JSONL format guidance for pretraining.

* **Chores**
  * Ignore local experiment tracking files (.wandb) in repository ignores.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/training_hub/algorithms/osft.py [hunk 1]
```diff
@@ -359,26 +402,40 @@ def execute_training(self, algorithm_params: dict[str, any]) -> any:
...
         training_args_pre['data_path'] = training_ready_data_path  # replaces raw data path with processed
 
+        # Construct PretrainingConfig for mini-trainer if pretraining mode
+        if algorithm_params.get('is_pretraining', False):
+            if (block_size := algorithm_params.get('block_size', None)) is None:
+                raise ValueError('block_size is required when is_pretraining=True')
+            pretraining_config = PretrainingConfig(
+                block_size=block_size,
+            )
+            training_args_pre['pretraining_config'] = pretraining_config
```
[the remark this answers is not in the document]
Comment (RobotSail) [reply src/training_hub/algorithms/osft.py]: This is true only for the instructlab-training backend, which handles its own data processing internally. The mini-trainer backend offloads the data processing to instructlab-training's API, so mini-trainer only loads the raw tokenized datasets.
